storage/rocksdb/ha_rocksdb.cc

```diff
@@ -9047,21 +9053,48 @@ void ha_rocksdb::get_auto_increment(ulonglong off, ulonglong inc,
   if (inc == 1 && off == 1) {
     // Optimization for the standard case where we are always simply
     // incrementing from the last position
-    *first_value = auto_incr++;
+    if ((*first_value= auto_incr) != std::numeric_limits<longlong>::max()) {
+      // Make sure we don't wrap around
+      auto_incr++;
+    }
   } else {
     // The next value can be more complicated if either `inc` or 'off' is not 1
     longlong new_val;
     longlong last_val = auto_incr;
 
     // Loop until we can correctly update the atomic value
     do {
+      if (((last_val - off) / inc) ==
+          (std::numeric_limits<longlong>::max() - off) / inc) {
+        // The 'last_val' value is already equal to or larger than the largest
+        // value in the sequence.  Continuing would wrap around (technically
+        // the behavior would be undefined).  What should we do?
+        // We could:
+        //   1) set the new value to the last possible number in our sequence
+        //      as described above.  The problem with this is that this
+        //      number could be smaller than a value in an existing row.
+        //   2) set the new value to the largest possible number.  This number
+        //      may not be in our sequence, but it is guaranteed to be equal
+        //      to or larger than any other value already inserted.
+        //
+        //  For now I'm going to take option @2.
+        new_val = std::numeric_limits<longlong>::max();
+        auto_incr = new_val; // Store the largest value into auto_incr
```

Comment:
Blindly setting auto_incr to max value can cause auto_incr to overflow when combined with concurrent calls to auto_incr++. Again, I think you need to perform a cas here to do this correctly.